src/test/java/teammates/test/cases/webapi/GetStudentProfileActionTest.java

```diff
@@ -28,80 +30,175 @@ protected String getRequestMethod() {
-        assertEquals(student.getName(), output.getName());
+        StudentProfileData actualProfile = (StudentProfileData) result.getOutput();
+        assertEquals(expectedName, actualProfile.getName());
+        assertEquals(expectedProfile.email, actualProfile.getEmail());
+        assertEquals(expectedProfile.shortName, actualProfile.getShortName());
+        assertEquals(expectedProfile.institute, actualProfile.getInstitute());
+        assertEquals(expectedProfile.moreInfo, actualProfile.getMoreInfo());
+        assertEquals(expectedProfile.nationality, actualProfile.getNationality());
+        assertEquals(expectedProfile.gender, actualProfile.getGender());
+        assertEquals(expectedProfile.pictureKey, actualProfile.getPictureKey());
     }
 
-    private void testActionInMasquerade(AccountAttributes student) {
-        gaeSimulation.loginAsAdmin("admin.user");
-
-        ______TS("Typical case: masquerade mode");
-        StudentProfileAttributes student1InCourse1Profile = typicalBundle.profiles.get("student1InCourse1");
-
+    @Test
+    public void testExecute_withProfileNotYetCreated_shouldReturnEmptyProfile() {
+        AccountAttributes student2InCourse1 = typicalBundle.accounts.get("student2InCourse1");
+        String expectedName = student2InCourse1.name;
+        StudentProfileAttributes expectedProfile = StudentProfileAttributes.builder(student2InCourse1.googleId).build();
+        loginAsStudent(student2InCourse1.googleId);
         String[] submissionParams = new String[] {
-                Const.ParamsNames.STUDENT_ID, student.googleId,
-                Const.ParamsNames.USER_ID, student.googleId,
+                Const.ParamsNames.STUDENT_ID, student2InCourse1.googleId,
         };
 
         GetStudentProfileAction action = getAction(submissionParams);
         JsonResult result = getJsonResult(action);
-
         assertEquals(HttpStatus.SC_OK, result.getStatusCode());
-        StudentProfile output = (StudentProfile) result.getOutput();
-        output.getStudentProfile().modifiedDate = null; // ignore this field for testing
-        assertEquals(student1InCourse1Profile.toString(), output.getStudentProfile().toString());
-        assertEquals(student.getName(), output.getName());
+        StudentProfileData actualProfile = (StudentProfileData) result.getOutput();
+        assertEquals(expectedName, actualProfile.getName());
+        assertEquals(expectedProfile.email, actualProfile.getEmail());
+        assertEquals(expectedProfile.shortName, actualProfile.getShortName());
+        assertEquals(expectedProfile.institute, actualProfile.getInstitute());
+        assertEquals(expectedProfile.moreInfo, actualProfile.getMoreInfo());
+        assertEquals(expectedProfile.nationality, actualProfile.getNationality());
+        assertEquals(expectedProfile.gender, actualProfile.getGender());
+        assertEquals(expectedProfile.pictureKey, actualProfile.getPictureKey());
     }
 
```

Comment:
Should add test cases for all 3 cases you mentioned.
1. student fetch own profile.
2. student fetch teammates's profile
3. instructor fetch student's profile 

And add test cases for the condition where account is both student and instructor, should be able to fetch profile as either role.

Reply:
Well, I think this is unnecessary because `execute()` is blind to who request the profile. All it does is to return the profile of the given google id, and we just need to make sure the function is called in integration test. Therefore I think testing these cases in `checkAccessControl()` is enough.

Reply:
Noted.